### ISSUE TITLE
400 error after merging branches

### ISSUE THREAD

**Issue by grachevn** [DefectDojo/django-DefectDojo#4083]
Hello,
**Bug description**

we're testing DefectDojo integration with Gitlab-CI in our company and I get strange problem with request to DefectDojo.

I use this job for create engagement in DefectDojo in .pre stage of pipeline and it working fine before I accept merge request

```
defectdojo_create_engagement:
  stage: .pre
  image: alpine
  variables:
    GIT_STRATEGY: none
  allow_failure: true
  rules:
    - if: '$DEFECTDOJO_NOT_ON_MASTER == "true" && $CI_COMMIT_BRANCH == "master"'
      when: never
    - when: always
  before_script:
    - apk add curl jq coreutils
    - TODAY=`date +%Y-%m-%d`
    - ENDDAY=$(date -d "+${DEFECTDOJO_ENGAGEMENT_PERIOD} days" +%Y-%m-%d)
  script:
    - |
        ENGAGEMENTID=`curl --fail --location --request POST "${DEFECTDOJO_URL}/engagements/" \
              --header "Authorization: Token ${DEFECTDOJO_TOKEN}" \
              --header 'Content-Type: application/json' \
                --data-raw "{
                  \"tags\": [\"GITLAB-CI\"],
                  \"name\": \"#${CI_PIPELINE_ID}\",
                  \"description\": \"${CI_COMMIT_DESCRIPTION}\",
                  \"version\": \"${CI_COMMIT_REF_NAME}\",
                  \"first_contacted\": \"${TODAY}\",
                  \"target_start\": \"${TODAY}\",
                  \"target_end\": \"${ENDDAY}\",
                  \"reason\": \"string\",
                  \"tracker\": \"${CI_PROJECT_URL}/-/issues\",
                  \"threat_model\": \"${DEFECTDOJO_ENGAGEMENT_THREAT_MODEL}\",
                  \"api_test\": \"${DEFECTDOJO_ENGAGEMENT_API_TEST}\",
                  \"pen_test\": \"${DEFECTDOJO_ENGAGEMENT_PEN_TEST}\",
                  \"check_list\": \"${DEFECTDOJO_ENGAGEMENT_CHECK_LIST}\",
                  \"status\": \"${DEFECTDOJO_ENGAGEMENT_STATUS}\",
                  \"engagement_type\": \"CI/CD\",
                  \"build_id\": \"${CI_PIPELINE_ID}\",
                  \"commit_hash\": \"${CI_COMMIT_SHORT_SHA}\",
                  \"branch_tag\": \"${CI_COMMIT_REF_NAME}\",
                  \"deduplication_on_engagement\": \"${DEFECTDOJO_ENGAGEMENT_DEDUPLICATION_ON_ENGAGEMENT}\",
                  \"product\": \"${DEFECTDOJO_PRODUCTID}\",
                  \"source_code_management_uri\": \"${CI_PROJECT_URL}\",
                  \"build_server\": ${DEFECTDOJO_ENGAGEMENT_BUILD_SERVER},
                  \"source_code_management_server\": ${DEFECTDOJO_ENGAGEMENT_SOURCE_CODE_MANAGEMENT_SERVER},
                  \"orchestration_engine\": ${DEFECTDOJO_ENGAGEMENT_ORCHESTRATION_ENGINE}
                }" | jq -r '.id'`
    - echo "DEFECTDOJO_ENGAGEMENTID=${ENGAGEMENTID}" >> defectdojo.env
  artifacts:
    reports:
      dotenv: defectdojo.env
```

after merging branches i get 400 error in response from DefectDojo until I commit anything in branch. After commit it starts working fine again. What could I be doing wrong?

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [X] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: CentOS Linux 7
 - DefectDojo Commit Message: [2021-03-07 18:05:30 +0100] 6a2b0e5: remove roadmap items [ (HEAD, origin/master, origin/HEAD, master)]

**Comment by valentijnscholten** [comment]
As said on slack we would need the response message from the API, debug logs from defect dojo, sample report causing the problem etc. There is no direct relationship between merging in gitlab and defect dojo, so probably something unrelated.

**Comment by grachevn** [comment]
Hi, Valentijn

Job log:
```
Running with gitlab-runner 13.9.0 (2ebc4dc4)
  on lnxcrucpuap1015 xBgCMGga
Preparing the "docker" executor 00:18
Using Docker executor with image alpine ...
Pulling docker image alpine ...
Using docker image sha256:28f6e27057430ed2a40dbdd50d2736a3f0a295924016e294938110eeb8439818 for alpine with digest alpine@sha256:a75afd8b57e7f34e4dad8d65e2c7ba2e1975c795ce1ee22fa34f8cf46f96a3be ...
Preparing environment 00:00
Running on runner-xbgcmgga-project-4-concurrent-0 via lnxcrucpuap1015...
Getting source from Git repository 00:01
Skipping Git repository setup
Skipping Git checkout
Skipping Git submodules setup
Executing "step_script" stage of the job script 00:02
Using docker image sha256:28f6e27057430ed2a40dbdd50d2736a3f0a295924016e294938110eeb8439818 for alpine with digest alpine@sha256:a75afd8b57e7f34e4dad8d65e2c7ba2e1975c795ce1ee22fa34f8cf46f96a3be ...
$ apk add curl jq coreutils
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
(1/13) Installing libacl (2.2.53-r0)
(2/13) Installing libattr (2.4.48-r0)
(3/13) Installing skalibs (2.10.0.0-r0)
(4/13) Installing s6-ipcserver (2.10.0.0-r0)
(5/13) Installing utmps (0.1.0.0-r0)
Executing utmps-0.1.0.0-r0.pre-install
(6/13) Installing coreutils (8.32-r2)
(7/13) Installing ca-certificates (20191127-r5)
(8/13) Installing brotli-libs (1.0.9-r3)
(9/13) Installing nghttp2-libs (1.42.0-r1)
(10/13) Installing libcurl (7.74.0-r1)
(11/13) Installing curl (7.74.0-r1)
(12/13) Installing oniguruma (6.9.6-r0)
(13/13) Installing jq (1.6-r1)
Executing busybox-1.32.1-r3.trigger
Executing ca-certificates-20191127-r5.trigger
OK: 10 MiB in 27 packages
$ TODAY=`date +%Y-%m-%d`
$ ENDDAY=$(date -d "+${DEFECTDOJO_ENGAGEMENT_PERIOD} days" +%Y-%m-%d)
$ ENGAGEMENTID=`curl --fail --location --request POST "${DEFECTDOJO_URL}/engagements/" \ # collapsed multi-line command
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   979    0     0  100   979      0   106k --:--:-- --:--:-- --:--:--  106k
curl: (22) The requested URL returned error: 400 Bad Request
Uploading artifacts for failed job 00:01
Uploading artifacts...
WARNING: defectdojo.env: no matching files         
ERROR: No files to upload                          
ERROR: Job failed: exit code 22
```

NGINX log:
`172.20.0.1 - - [19/Mar/2021:06:05:32 +0000] "POST /api/v2/engagements/ HTTP/1.1" 400 88 "-" "curl/7.74.0" "-"`

DefectDojo logs:
```
*** Operational MODE: preforking+threaded ***
[19/Mar/2021 10:29:45] INFO [dojo.models:3360] enabling audit logging
WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x564444411320 pid: 1 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (pid: 1, cores: 2)
spawned uWSGI worker 2 (pid: 24, cores: 2)
[pid: 24|app: 0|req: 1/1] 172.20.0.1 () {42 vars in 1165 bytes} [Fri Mar 19 10:34:51 2021] GET / => generated 0 bytes in 86 msecs (HTTP/1.1 302) 7 headers in 215 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 2/2] 172.20.0.1 () {42 vars in 1188 bytes} [Fri Mar 19 10:34:51 2021] GET /login?next=/ => generated 13346 bytes in 3365 msecs (HTTP/1.1 200) 9 headers in 472 bytes (1 switches on core 1)
/usr/local/lib/python3.6/site-packages/jose/backends/cryptography_backend.py:18: CryptographyDeprecationWarning: int_from_bytes is deprecated, use int.from_bytes instead
  from cryptography.utils import int_from_bytes, int_to_bytes
[pid: 24|app: 0|req: 3/3] 172.20.0.1 () {50 vars in 1467 bytes} [Fri Mar 19 10:35:11 2021] POST /login?next=/ => generated 0 bytes in 376 msecs (HTTP/1.1 302) 11 headers in 629 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 4/4] 172.20.0.1 () {46 vars in 1371 bytes} [Fri Mar 19 10:35:11 2021] GET / => generated 0 bytes in 48 msecs (HTTP/1.1 302) 7 headers in 212 bytes (1 switches on core 1)
[pid: 24|app: 0|req: 5/5] 172.20.0.1 () {46 vars in 1389 bytes} [Fri Mar 19 10:35:12 2021] GET /dashboard => generated 41405 bytes in 300 msecs (HTTP/1.1 200) 7 headers in 336 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 6/6] 172.20.0.1 () {44 vars in 1300 bytes} [Fri Mar 19 10:35:22 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 24|app: 0|req: 7/7] 172.20.0.1 () {44 vars in 1300 bytes} [Fri Mar 19 10:35:33 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 8/8] 172.20.0.1 () {44 vars in 1300 bytes} [Fri Mar 19 10:35:43 2021] GET /alerts/count => generated 13 bytes in 10 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 24|app: 0|req: 9/9] 172.20.0.1 () {44 vars in 1300 bytes} [Fri Mar 19 10:35:53 2021] GET /alerts/count => generated 13 bytes in 10 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 10/10] 172.20.0.1 () {44 vars in 1300 bytes} [Fri Mar 19 10:36:03 2021] GET /alerts/count => generated 13 bytes in 9 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 24|app: 0|req: 11/11] 172.20.0.1 () {44 vars in 1350 bytes} [Fri Mar 19 10:36:13 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 12/12] 172.20.0.1 () {44 vars in 1393 bytes} [Fri Mar 19 10:36:24 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 1|app: 0|req: 1/13] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:36:40 2021] GET /alerts/count => generated 13 bytes in 2398 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 24|app: 0|req: 13/14] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:37:00 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
**[pid: 24|app: 0|req: 14/15] 172.20.0.1 () {38 vars in 528 bytes} [Fri Mar 19 10:37:01 2021] POST /api/v2/engagements/ => generated 88 bytes in 12 msecs (HTTP/1.1 400) 7 headers in 222 bytes (1 switches on core 1)**
[pid: 24|app: 0|req: 15/16] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:37:26 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 1|app: 0|req: 2/17] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:38:20 2021] GET /alerts/count => generated 13 bytes in 11 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 1|app: 0|req: 3/18] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:39:29 2021] GET /alerts/count => generated 13 bytes in 10 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
[pid: 1|app: 0|req: 4/19] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:40:11 2021] GET /alerts/count => generated 13 bytes in 10 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 1)
[pid: 1|app: 0|req: 5/20] 172.20.0.1 () {44 vars in 1442 bytes} [Fri Mar 19 10:42:39 2021] GET /alerts/count => generated 13 bytes in 9 msecs (HTTP/1.1 200) 6 headers in 180 bytes (1 switches on core 0)
```

There is nothing in logs I can see.
I enable DEBUG in settings.dist.py, right?

**Comment by grachevn** [comment]
I found problem, it's in ${CI_COMMIT_DESCRIPTION} variable.
First, I use netcat for capture curl request from gitlab. Description field value while we committing to gitlab is "", but after merging branches it value 
```
"description": "
Update gitlab-defectdojo.yml

See merge request k8s-cicd/sandbox!17"
```

and it brakes request to API.

Second, I removed` \"description\": \"${CI_COMMIT_DESCRIPTION}\"` line from pipeline and it start working after merge request.
